packages/core/android/src/main/java/com/openmobilehub/reactnative/auth/core/OmhAuthModuleImpl.kt

```diff
@@ -15,44 +15,34 @@
 
 package com.openmobilehub.reactnative.auth.core
 
+import android.app.Activity
 import com.facebook.react.bridge.Arguments
 import com.facebook.react.bridge.Promise
 import com.facebook.react.bridge.ReactApplicationContext
-import com.facebook.react.bridge.ReactContextBaseJavaModule
-import com.facebook.react.bridge.ReactMethod
 import com.facebook.react.bridge.ReadableMap
 import com.openmobilehub.android.auth.core.OmhAuthClient
 
-class OmhAuthModule(
-    reactContext: ReactApplicationContext,
-    private val name: String,
-    private val createOmhAuthClient: (config: HashMap<String, Any>) -> OmhAuthClient
-) : ReactContextBaseJavaModule(reactContext) {
-    private var omhAuthClient: OmhAuthClient? = null
-    private fun getAuthClient(): OmhAuthClient {
-        if (omhAuthClient == null) {
-            throw Exception("OmhAuthClient is not initialized")
-        }
+class OmhAuthModuleImpl(
+    context: ReactApplicationContext,
+    private val createAuthClient: (config: HashMap<String, Any>) -> OmhAuthClient
+) {
+    private var _authClient: OmhAuthClient? = null
+
+    val authClient: OmhAuthClient
+        get() = _authClient ?: throw Exception("OmhAuthClient is not initialized")
```

Comment:
```suggestion
        get() = _authClient ?: throw IllegalStateException("Accessing OmhAuthClient before OmhAuthModule is initialized")
```

Reply:
Here I wanted to do the exception and message a bit more specific to the problem - if we would be satisfied with general exception without specific message then we could go with late init variable like:
```
    lateinit var authClient: OmhAuthClient
        private set
```

Reply:
Fixed according to the suggestion.